pkg/services/govmomi/service.go

```diff
@@ -54,27 +55,27 @@ type VMService struct{}
 //  2. Updating the VM with the bootstrap data, such as the cloud-init meta and user data, before...
 //  3. Powering on the VM, and finally...
 //  4. Returning the real-time state of the VM to the caller
-func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMachine, _ error) {
+func (vms *VMService) ReconcileVM(ctx context.Context, vmCtx *capvcontext.VMContext) (vm infrav1.VirtualMachine, _ error) {
 	// Initialize the result.
 	vm = infrav1.VirtualMachine{
-		Name:  ctx.VSphereVM.Name,
+		Name:  vmCtx.VSphereVM.Name,
 		State: infrav1.VirtualMachineStatePending,
 	}
 
 	// If there is an in-flight task associated with this VM then do not
 	// reconcile the VM until the task is completed.
-	if inFlight, err := reconcileInFlightTask(ctx); err != nil || inFlight {
+	if inFlight, err := reconcileInFlightTask(vmCtx); err != nil || inFlight {
 		return vm, err
 	}
 
 	// This deferred function will trigger a reconcile event for the
 	// VSphereVM resource once its associated task completes. If
 	// there is no task for the VSphereVM resource then no reconcile
 	// event is triggered.
-	defer reconcileVSphereVMOnTaskCompletion(ctx)
+	defer reconcileVSphereVMOnTaskCompletion(vmCtx)
 
 	// Before going further, we need the VM's managed object reference.
-	vmRef, err := findVM(ctx)
+	vmRef, err := findVM(vmCtx)
```

Comment:
```suggestion
	vmRef, err := findVM(ctx, vmCtx)
```
Need to refactor func https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/pkg/services/govmomi/util.go#L60